c10/core/Allocator.h

```diff
@@ -230,6 +231,19 @@ struct C10_API InefficientStdFunctionContext {
   std::function<void(void*)> deleter_;
   InefficientStdFunctionContext(void* ptr, std::function<void(void*)> deleter)
       : ptr_(ptr), deleter_(std::move(deleter)) {}
+  InefficientStdFunctionContext(const InefficientStdFunctionContext&) = delete;
+  InefficientStdFunctionContext(InefficientStdFunctionContext&& rhs) noexcept
```

Comment:
Why can't this remain default? Just for the nullptr invalidation?

Reply:
It should be deleted because the destructor uses the deleter. It's more like std::unique_ptr.

Reply:
It should be deleted because the destructor used deleter, just like std::unique_ptr.